pkg/utils/utils_test.go

```diff
@@ -421,3 +713,2142 @@ func TestSetContainerImage(t *testing.T) {
+	// Test case: Valid YAML - DaemonSet
+	// TODO: This string does not cover all elements necessary to build the rbac object.
+	yamlString := `
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: my-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list"]
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  key: value
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: my-daemonset
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: MyApp
+  template:
+    metadata:
+      labels:
+        app: MyApp
+    spec:
+      containers:
+      - name: my-container
+        image: my-image
+        ports:
+        - containerPort: 8080
+    `
+
+	ctrlObject, err := GetDriverYaml(yamlString, "DaemonSet")
+	appsv1 := "apps/v1"
+	daemonset := "DaemonSet"
+	myDaemonset := "my-daemonset"
+	myContainer := "my-container"
+	myImage := "my-image"
+	port8080 := int32(8080)
+	// TODO: This object will need to be built precisely to match the input spec.
+	expected := NodeYAML{
+		DaemonSetApplyConfig: confv1.DaemonSetApplyConfiguration{
+			TypeMetaApplyConfiguration: confmetav1.TypeMetaApplyConfiguration{
+				APIVersion: &appsv1,
+				Kind:       &daemonset,
+			},
+			ObjectMetaApplyConfiguration: &confmetav1.ObjectMetaApplyConfiguration{
+				Name: &myDaemonset,
+			},
+			Spec: &confv1.DaemonSetSpecApplyConfiguration{
+				// Spec configuration
+				Selector: &confmetav1.LabelSelectorApplyConfiguration{
+					MatchLabels: map[string]string{
+						"app": "MyApp",
+					},
+				},
+				Template: &confcorev1.PodTemplateSpecApplyConfiguration{
+					// Template configuration
+					Spec: &confcorev1.PodSpecApplyConfiguration{
+						Containers: []confcorev1.ContainerApplyConfiguration{
+							{
+								// Container configuration
+								Name:  &myContainer,
+								Image: &myImage,
+								Ports: []confcorev1.ContainerPortApplyConfiguration{
+									{
+										ContainerPort: &port8080,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Status: &confv1.DaemonSetStatusApplyConfiguration{
+				// Status configuration
+			},
+		},
+		Rbac: RbacYAML{
+			ServiceAccount: corev1.ServiceAccount{
+				// ServiceAccount configuration
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "ServiceAccount",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "my-service",
+				},
+			},
+			ClusterRole: rbacv1.ClusterRole{
+				// ClusterRole configuration
+			},
+			ClusterRoleBinding: rbacv1.ClusterRoleBinding{
+				// ClusterRoleBinding configuration
+			},
+		},
+	}
+
+	assert.Nil(t, err)
+	// TODO: Proper comparison here once the
+	// expected object has been ironed out
+	assert.NotNil(t, ctrlObject)
+	assert.NotNil(t, expected)
```

Comment:
Is this really a sufficient test?

Reply:
Added assert to few attributes.. Asserted the hardcoded replicas with the status object attributes from expected object.
Can be further improved to remove the hardcoded replicas number.